packages/filesystem/src/browser/file-tree/file-tree-model.ts

```diff
@@ -111,6 +110,17 @@ export class FileTreeModel extends TreeModel implements LocationService {
         return true;
     }
 
+    /**
+     * Move the given source file or directory to the given target directory.
+     */
+    async move(source: ITreeNode, target: ITreeNode) {
+        if (DirNode.is(target) && FileStatNode.is(source)) {
+            const sourceUri = source.uri.toString();
+            const targetUri = target.uri.resolve(source.name).toString();
+            await this.fileSystem.move(sourceUri, targetUri, { overwrite: true });
```

Comment:
to workaround https://github.com/theia-ide/theia/pull/975#issuecomment-352356821 hereafter `move`:
```ts
// to workaround https://github.com/Axosoft/nsfw/issues/42
this.refresh(target);
```